### PR TITLE
Refactor PORO creation and MovieService

### LINKED ISSUE #34: Viewing Parties: Add update and delete
Quality of life update

### LINKED ISSUE #43: Refactor Movie Poro
Move credits and reviews out of Movie Poro and let `MoviesFacade` take that functionality of creating those objects

### PR DESCRIPTION
**Pull request by ETBassist**
- Move CastMember instantiation to MoviesFacade
- Move Review instantiation to MoviesFacade
- Remove superfluous reviews_count from Movie PORO
- Add vote_average to movie show page
- Refactor viewing party controller for Rubocop
- Add dependent: :destroy to user and party models
- Refactor MovieService top_rated and by_keyword

____ Wrote Tests X____ Implemented ____ Reviewed

Neccesary checkmarks:

    [x] All Tests are Passing

    [x] The code will run locally

Type of change

    [] New feature
    [] Bug Fix

Implements/Fixes:

Closes #43

Check the correct boxes

    [x] This broke nothing
    [] This broke some stuff
    [] This broke everything

Testing Changes

    [] No Tests have been changed
    [x] Some Tests have been changed
    [] All of the Tests have been changed(Please describe what in the world happened)

Checklist:

    [x] My code has no unused/commented out code
    [] I have reviewed my code
    [] I have commented my code, particularly in hard-to-understand areas
    [x] I have fully tested my code

(For Fun!)Please Include a link to a gif of your feelings about this branch

Link:

### CODE REVIEW

**Review by Jonathan-M-Wilson — APPROVED** [review]
Looks amazing! Great work!

File: app/controllers/movies_controller.rb [hunk 1]
```diff
@@ -9,5 +9,7 @@ def index
 
   def show
     @movie = MoviesFacade.movie(params[:id])
+    @top_ten_cast_members = MoviesFacade.cast_members_for(@movie)
+    @reviews = MoviesFacade.reviews_for(@movie)
```
Comment (helloeduardo) [inline app/controllers/movies_controller.rb:13]: Nice solution to Dione's suggestion

File: app/controllers/viewing_party_controller.rb [hunk 2]
```diff
@@ -30,9 +20,48 @@ def create
...
+      render :edit
+    end
+  end
+
+  def delete
+    party = Party.find(params[:id])
+    Invitation.where(party_id: party.id).destroy_all
+    party.destroy
+    redirect_to "/dashboard"
+  end
```
Comment (helloeduardo) [inline app/controllers/viewing_party_controller.rb:50]: Thanks for adding these CRUD actions

File: app/controllers/viewing_party_controller.rb [hunk 3]
```diff
@@ -30,9 +20,48 @@ def create
...
+
+  def create_and_send_invite_to(friend_id, party)
+    friend = User.find(friend_id)
+    email_info = { user: current_user,
+                   friend: friend,
+                   message: party.movie_title }
+    Invitation.create(party_id: party.id,
+                      user_id: friend_id)
+    ViewingPartyMailer.inform(email_info, friend.email).deliver_now
+  end
```
Comment (helloeduardo) [inline app/controllers/viewing_party_controller.rb:66]: Nice job extracting that method

File: app/models/user.rb [hunk 4]
```diff
@@ -7,6 +7,6 @@ class User < ApplicationRecord
   has_many :friendships, dependent: :destroy
   has_many :friends, through: :friendships
 
-  has_many :invitations
+  has_many :invitations, dependent: :destroy
```
Comment (helloeduardo) [inline app/models/user.rb:10]: Nice catch with the destroy

**Review by helloeduardo — APPROVED** [review]
Thanks for these improvements!

**Pull request merged**